iib/web/errors.py

```diff
@@ -32,3 +34,45 @@ def json_error(error):
         response = jsonify({'error': msg})
         response.status_code = status_code
     return response
+
+
+def handle_broker_error(request_id, request_model):
```

Comment:
Could you please just accept the request object so that you don't have to query the database for it?